### PR TITLE
Update to GCP k8s and new Address Index API

### PR DESCRIPTION
**Pull request by bambrp**
## Context?
We have a requirement for a proxy layer to serve up possible postal addresses from postcode or partial text matches from a lookup service.

This service was designed to test a specific internal provider, that was only available if we connected to it via VPC peering on the Google Cloud Platform, which necessitated the creation of kubernetes deployment scripts.

There was also a need to update the API calls to the latest version, and to restrict the number of address suggestions returned to a more reasonable 10 (down from 100).

## Reviewing
The onward service this proxy was created to talk to is hosted on a GCP project that you must enable VPC sharing with.

Additionally, since that service had no auth of its own, the proxy was responsible for securing it- this is done through GCP's Cloud Armor applying a set of firewall rules (a policy called _ons-only_) to restrict to a set of known IP addresses. You will need to implement this policy to make this deployable, or remove the rule in the BackendConfig section of _k8s/helm/templates/service.yaml_

Deployment is by running `k8s/deploy_app.sh LOOKUP_API_URL DOCKER_REGISTRY IMAGE_TAG` from the top level folder.

### CODE REVIEW

File: app/request_address.py [hunk 1]
```diff
@@ -12,44 +12,43 @@
 
 
 def query_address_index(search_value):
-    '''
+    """
```
Comment (JamesGardiner) [inline app/request_address.py:15]: Runner uses single quoted strings - we should maintain consistency across apps to make tooling simpler

File: app/request_address.py [hunk 2]
```diff
@@ -12,44 +12,43 @@
...
         # Postcode search
         request_string = "/".join((settings.LOOKUP_URL, 'addresses', 'postcode', search_value))
     else:
         request_string = "/".join(
-            (settings.LOOKUP_URL, 'addresses', 'partial', search_value + '?limit=' + settings.RESULT_LIMIT))
-
-    headers = {
-        'Authorization': settings.AUTH_KEY
-    }
+            (settings.LOOKUP_URL, 'addresses', 'partial?input=' + search_value + '&limit=' + settings.RESULT_LIMIT))
```
Comment (JamesGardiner) [inline app/request_address.py]: Could you break this up over multiple line to make it more readable?

File: app/request_address.py [hunk 3]
```diff
@@ -12,44 +12,43 @@
...
                                  settings.REQUEST_READ_TIMEOUT),
+                        auth=(settings.AUTH_KEY, ''),
                         headers=headers)
     g.address_index_response_time = time.time() - address_index_request_start
     logger.bind(address_index_response_time=g.address_index_response_time)
 
     if resp.status_code != 200:
         # This means something went wrong.
-        raise Exception("ERROR IN ADDRESS INDEX API (try refresh). Reason: {}. Response: {}".format(resp.reason, resp.text))
+        raise Exception("ERROR IN ADDRESS INDEX API (try refresh). Reason: {}. Response: {}".format(resp.reason,
```
Comment (JamesGardiner) [inline app/request_address.py]: This should be an f-string

**Review by JamesGardiner — CHANGES_REQUESTED** [review]
Some minor formatting suggestions

File: app/request_address.py (hunk 1 above)
Comment (bambrp) [reply app/request_address.py:15]: I altered this as it's recommended by PEP257 that says docstrings should always be triple double-quoted. I have no horse in that race, will change if you want.

File: app/request_address.py (hunk 2 above)
Comment (bambrp) [reply app/request_address.py]: What's the current agreed line length with you guys? PEP8's 79 chars, a bit more?

Comment (JamesGardiner) [reply app/request_address.py]: If we go down the Black formatting route (which we are in Runner), then 88

File: app/request_address.py (hunk 3 above)
Comment (bambrp) [reply app/request_address.py]: I tried to leave as much alone as I could and have it work, and only fixed things the PyCharm linter was moaning about. You want all strings of this ilk converting? i.e. template strings with variables to expand into them

Comment (JamesGardiner) [reply app/request_address.py]: Ok, if we are being light touch then it seems fair to leave it as is, and make a wholesale change to f-strings in future (should we decide to in this project)

File: app/request_address.py (hunk 2 above)
Comment (bambrp) [reply app/request_address.py]: 88, got it.

**Review by JamesGardiner — APPROVED** [review]

**Comment by JamesGardiner** [comment]
@bambrp @pricem14pc can this be closed?

**Comment by pricem14pc** [comment]
This can be re-visited when the AI requirement resurfaces.

**Pull request closed without merging**